pulp_container/tests/functional/api/test_rbac_repo_content.py

```diff
@@ -0,0 +1,217 @@
+            [
+                "container.view_containerrepository",
+                "container.view_containerpushrepository",
+            ]
+        )
+        cls.user_reader2 = gen_user(["container.view_containerrepository"])
+        cls.user_reader3 = gen_user(["container.view_containerpushrepository"])
+        cls.user_helpless = gen_user([])
+
+        # create a push repo with user_creator
+        image_path = f"{DOCKERHUB_PULP_FIXTURE_1}:manifest_a"
+        cls._pull(image_path)
+        repo_name = "testcontent/perms"
+        local_url = "/".join([cls.registry_name, f"{repo_name}:1.0"])
+        cls._push(image_path, local_url, cls.user_creator)
+        cls.push_repository = cls.pushrepository_api.list(name=repo_name).results[0]
+
+        # create a second push repo with user_creator2
+        image_path = f"{DOCKERHUB_PULP_FIXTURE_1}:manifest_b"
+        cls._pull(image_path)
+        repo_name = "testcontent2/perms"
+        local_url = "/".join([cls.registry_name, f"{repo_name}:1.0"])
+        cls._push(image_path, local_url, cls.user_creator2)
+        cls.push_repository2 = cls.pushrepository_api.list(name=repo_name).results[0]
+
+        # sync a repo with user_creator
+        cls.repository = cls.user_creator["repository_api"].create(
+            ContainerContainerRepository(**gen_repo())
+        )
+        cls.remote = cls.user_creator["remote_api"].create(
+            gen_container_remote(upstream_name=DOCKERHUB_PULP_FIXTURE_1)
+        )
+        sync_data = RepositorySyncURL(remote=cls.remote.pulp_href)
+        sync_response = cls.user_creator["repository_api"].sync(cls.repository.pulp_href, sync_data)
+        monitor_task(sync_response.task)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete api users and objects created in setUpclass."""
+        cls.user_creator["repository_api"].delete(cls.repository.pulp_href)
+        cls.user_creator["remote_api"].delete(cls.remote.pulp_href)
+        del_user(cls.user_creator)
+        del_user(cls.user_creator2)
+        del_user(cls.user_reader)
+        del_user(cls.user_reader2)
+        del_user(cls.user_reader3)
+        del_user(cls.user_helpless)
+        for name in ("testcontent", "testcontent2"):
+            namespace = cls.namespace_api.list(name=name).results[0]
+            cls.namespace_api.delete(namespace.pulp_href)
+        delete_orphans()
+
+    def test_content_list(self):
+        """
+        Test that users can list content if they have enough rights
+        """
+        push_repository_rv = self.pushrepository_api.read(
+            self.push_repository.pulp_href
+        ).latest_version_href
+        push_repository2_rv = self.pushrepository_api.read(
+            self.push_repository2.pulp_href
+        ).latest_version_href
+        repository_rv = self.repository_api.read(self.repository.pulp_href).latest_version_href
+
+        self.assertEqual(self.user_creator["tags_api"].list().count, 10)
```

Comment:
I tested tags only in this test, let me know if i should also add blobs/manifests. I found adding them redundant since they user same policy and queryset

Reply:
Sounds good to me.